src/sankey/index.js

```diff
@@ -0,0 +1,137 @@
+import React, {PropTypes, Component} from 'react';
+import {voronoi} from 'd3-voronoi';
+import {sankey} from 'd3-sankey-align';
+
+import {DISCRETE_COLOR_RANGE} from 'theme';
+
+const noop = f => f;
+
+const defaultLinkColor = DISCRETE_COLOR_RANGE[1];
+const defaultLinkOpacity = 0.7;
+const defaultNodeColor = DISCRETE_COLOR_RANGE[0];
+const defaultNodeOpacity = 1;
+
+class Sankey extends Component {
+
+  static defaultProps = {
+    margin: 20,
+    nodeWidth: 10,
+    nodePadding: 10,
+    align: 'justify',
+    layout: 50,
+    hasVoronoi: false,
+    onClick: noop,
+    onHover: noop,
+    onBlur: noop
+  }
+
+  static propTypes = {
+    width: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+    margin: PropTypes.number,
+    nodeWidth: PropTypes.number,
+    nodePadding: PropTypes.number,
+    layout: PropTypes.number,
+    align: PropTypes.oneOf(['justify', 'left', 'right', 'center']),
+    nodes: PropTypes.arrayOf(PropTypes.object).isRequired,
+    links: PropTypes.arrayOf(PropTypes.shape({
+      source: PropTypes.number.isRequired,
+      target: PropTypes.number.isRequired
+    })).isRequired,
+    hasVoronoi: PropTypes.bool,
+    onClick: PropTypes.func,
+    onHover: PropTypes.func,
+    onBlur: PropTypes.func
+  }
+
+  render() {
+
+    const {
+      width,
+      height,
+      margin,
+      nodeWidth,
+      nodePadding,
+      align,
+      layout,
+      nodes,
+      links,
+      hasVoronoi,
+      onClick,
+      onHover,
+      onBlur
+    } = this.props;
+
+    const sankeyInstance = sankey()
+      .size([width, height])
+      .nodeWidth(nodeWidth)
+      .nodePadding(nodePadding)
+      .nodes(nodes)
+      .links(links)
+      .align(align)
+      .layout(layout);
+
+    const nWidth = sankeyInstance.nodeWidth();
+    const path = sankeyInstance.link();
+
+    const voronoiInstance = voronoi()
+      .x(d => d.x + d.dx / 2)
+      .y(d => d.y + d.dy / 2)
+      .extent([[-margin, -margin], [width + margin, height + margin]]);
+
+    return (
+      <svg height={height + margin} width={width + margin} className="rv-sankey">
+        <g transform={`translate(${margin / 2}, ${margin / 2})`}>
+
+          {links.map((link, i) => (
+            <path
+              d={path(link)}
+              className="rv-sankey__link"
+              opacity={link.opacity || defaultLinkOpacity}
+              stroke={link.color || defaultLinkColor}
+              strokeWidth={Math.max(1, link.dy)}
+              fill="none"
+              id={link.id}
+              key={link.id || link.key || `link-${i}`} />
+          ))}
+
+          {nodes.map((node, i) => (
+            <g
+              transform={`translate(${node.x}, ${node.y})`}
+              className="rv-sankey__node"
+              opacity={node.opacity || defaultNodeOpacity}
```

Comment:
make sure that node.opacity can be zero, eg if you have a situation like

x = 0 || defaultNodeOpacity

x should be zero